streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -214,9 +214,18 @@ public void process(final Record<KIn, VIn> record) {
                 internalProcessorContext.currentNode().name(),
                 internalProcessorContext.taskId());
 
-            final ProcessingExceptionHandler.ProcessingHandlerResponse response = processingExceptionHandler
-                .handle(errorHandlerContext, record, e);
-
+            final ProcessingExceptionHandler.ProcessingHandlerResponse response;
+
+            try {
+                response = processingExceptionHandler
+                        .handle(errorHandlerContext, record, e);
+            } catch (final Exception fatalUserException) {
```

Comment:
Could you please also add a unit test for this?

Reply:
done